USAGE.md

```diff
@@ -1,30 +1,203 @@
+    - [Using devcontainers from the terminal](#using-devcontainers-from-the-terminal)
+    - [Generated build scripts](#generated-build-scripts)
+    - [rapids-build-utils](#rapids-build-utils)
+    - [Native build tools - CMake, python builds](#native-build-tools---cmake-python-builds)
+    - [Exiting the devcontainer](#exiting-the-devcontainer)
+  - [Adding projects: `manifest.yaml` file](#adding-projects-manifestyaml-file)
+  - [Using the pre-built images](#using-the-pre-built-images)
+    - [Using in `devcontainer.json`](#using-in-devcontainerjson)
+    - [Custom devcontainers](#custom-devcontainers)
+  - [Build caching with `sccache`](#build-caching-with-sccache)
+    - [Build caching with private S3 buckets](#build-caching-with-private-s3-buckets)
+    - [Using GitHub OAuth to issue S3 credentials via Hashicorp Vault](#using-github-oauth-to-issue-s3-credentials-via-hashicorp-vault)
+
+## Quick start
+
+If you really want to get started right away and ignore all the details of how
+devcontainers work, the easiest way is to treat the devcontainer
+like a docker container that you use interactively. Brad Dice wrote [a script
+that wraps the devcontainer
+CLI](https://gist.github.com/bdice/a92d224b3e3b1b387fc18b8095b3bdbd) to do this.
+
+To obtain this script:
+```
+curl -LO https://gist.githubusercontent.com/bdice/a92d224b3e3b1b387fc18b8095b3bdbd/raw/28eb8edc856ae04d4cd83571fea5b894f714f01c/rapids-dev
+chmod +x rapids-dev
+./rapids-dev
+```
+
+This script expects you to have your current directory set to the
+root of a repo that has a .devcontainer folder. You may want to move
+this script to a folder that you place on PATH, such as `~/bin`
+
+Running that command will build the devcontainer and drop you at
+an interactive prompt. You can immediately build your project
+with one of the devcontainers scripts. Type `build-` and hit `<TAB>` to see your options.
+
+Skip ahead to [Available tools in the devcontainer](#available-tools-in-the-devcontainer) to see more options
+for interaction at this prompt.
+
+## Detailed start
+
+So, you have cloned a repo that you've heard has a devcontainer. You can see the file(s) for yourself by looking
+in your repo's `.devcontainer` folder. You may find a `devcontainer.json` file, or you may find some number of folders, such as `cuda12.0-conda` and `cuda12.0-pip`. If you find folders, these each contain a `devcontainer.json`
+file. These files specify how to create and run a container that leaves you with a good setup to do development.
+
+There are at least 2 ways to consume these devcontainer.json files. VS Code was
+the original home of devcontainers prior to becoming an open specification, and
+it remains a good way to use them. If you prefer not to use VS Code, the
+[devcontainers-cli](https://code.visualstudio.com/docs/devcontainers/devcontainer-cli)
+application will build the devcontainer and allow you to interact with it.
+
+### Using devcontainers in VS Code
```

Comment:
Is this localhost only? or can I run devcontainers in VS and using the remote ssh plugin to get to my GPU workstation?

Reply:
Yes you can run them via the remote-ssh plugin.